aggregate/aggregator.go

```diff
@@ -19,31 +19,46 @@ var (
 )
 
 const (
-	maximumRecordSize       = 1024 * 1024 // 1 MB
-	defaultMaxAggRecordSize = 20 * 1024   // 20K
-	initialAggRecordSize    = 0
-	fieldNumberSize         = 1 // All field numbers are below 16, meaning they will only take up 1 byte
+	defaultMaximumRecordSize = 1024 * 1024 // 1 MB
+	defaultMaxAggRecordSize  = 20 * 1024   // 20K
+	initialAggRecordSize     = 0
+	fieldNumberSize          = 1 // All field numbers are below 16, meaning they will only take up 1 byte
 )
 
 // Aggregator kinesis aggregator
 type Aggregator struct {
-	partitionKeys    map[string]uint64
-	records          []*Record
-	aggSize          int // Size of both records, and partitionKeys in bytes
-	maxAggRecordSize int
-	stringGen        *util.RandomStringGenerator
+	partitionKeys     map[string]uint64
+	records           []*Record
+	aggSize           int // Size of both records, and partitionKeys in bytes
+	maximumRecordSize int
+	maxAggRecordSize  int
+	stringGen         *util.RandomStringGenerator
 }
 
-// NewAggregator create a new aggregator
-func NewAggregator(stringGen *util.RandomStringGenerator) *Aggregator {
+// Config is for aggregation customizations.
+type Config struct {
+	MaximumRecordSize *int
```

Comment:
OOC is the convention different for this kind of struct ? other variables are in camelCase

Reply:
Do you mean starting with a capital character? That is to expose variables to other packages to be able to set it. `kinesis package sets it.